sway-core/src/asm_generation/fuel/fuel_asm_builder.rs

```diff
@@ -1249,6 +1250,40 @@ impl<'ir, 'eng> FuelAsmBuilder<'ir, 'eng> {
         Ok(())
     }
 
+    fn compile_get_global(
+        &mut self,
+        instr_val: &Value,
+        local_var: &GlobalVar,
+    ) -> Result<(), CompileError> {
+        let span = self
+            .md_mgr
+            .val_to_span(self.context, *instr_val)
+            .unwrap_or_else(Span::dummy);
+        let Some(constant) = local_var.get_initializer(self.context) else {
+            return Err(CompileError::Internal(
+                "Global variable must have an initializer.",
```

Comment:
As also tackled in other comments, if we model the `GlobalVar` with the same semantics as the local vars which we aim for, then this doesn't need to be true. We can have `mut` global vars that are not necessarily initialized. And as mentioned above, those should not be in the data section.